src/model/application/interaction.rs

```diff
@@ -289,11 +293,86 @@ bitflags! {
+///
+/// [Discord Docs](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-authorizing-integration-owners-object)
+#[cfg_attr(feature = "typesize", derive(typesize::derive::TypeSize))]
+#[cfg(feature = "unstable_discord_api")]
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub enum AuthorizingIntegrationOwner {
+    /// The [`Application`] was installed to a guild, containing the id if invoked in said guild.
+    ///
+    /// [`Application`]: super::CurrentApplicationInfo
+    GuildInstall(Option<GuildId>),
+    /// The [`Application`] was installed to a user, containing the id of said user.
+    ///
+    /// [`Application`]: super::CurrentApplicationInfo
+    UserInstall(UserId),
+    Unknown(InstallationContext),
+}
+
+#[cfg(feature = "unstable_discord_api")]
+impl<'de> serde::Deserialize<'de> for AuthorizingIntegrationOwner {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> StdResult<Self, D::Error> {
+        struct Visitor;
+
+        impl<'de> serde::de::Visitor<'de> for Visitor {
+            type Value = AuthorizingIntegrationOwner;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                formatter.write_str("a hashmap containing keys of InstallationContext and values based on those keys")
+            }
+
+            fn visit_map<A>(self, mut map: A) -> StdResult<Self::Value, A::Error>
+            where
+                A: serde::de::MapAccess<'de>,
+            {
+                use crate::model::utils::CowStr;
+
+                let key_str = map
+                    .next_key::<CowStr<'_>>()?
+                    .ok_or_else(|| serde::de::Error::missing_field("key"))?;
+
+                let key_int = key_str.0.parse::<u8>().map_err(serde::de::Error::custom)?;
+                match InstallationContext::from(key_int) {
+                    InstallationContext::Guild => map.next_value().map(Self::Value::GuildInstall),
+                    InstallationContext::User => map.next_value().map(Self::Value::UserInstall),
+                    key => Ok(Self::Value::Unknown(key)),
+                }
+            }
+        }
+
+        deserializer.deserialize_map(Visitor)
+    }
+}
+
+#[cfg(feature = "unstable_discord_api")]
+impl serde::Serialize for AuthorizingIntegrationOwner {
+    fn serialize<S: Serializer>(&self, serializer: S) -> StdResult<S::Ok, S::Error> {
+        use serde::ser::SerializeMap;
+
+        let mut serializer = serializer.serialize_map(Some(1))?;
+        match self {
+            Self::GuildInstall(inner) => {
+                serializer.serialize_entry(&InstallationContext::Guild, &inner)
+            },
+            Self::UserInstall(inner) => {
+                serializer.serialize_entry(&InstallationContext::User, &inner)
+            },
+            Self::Unknown(inner) => serializer.serialize_entry(&inner, &()),
+        }?;
+
+        serializer.end()
+    }
+}
```

Comment:
A response can contain multiple (2) authorizing integration owners. I'm not sure what the best way to go about representing it would be. I'm currently experimenting with having something like this
```rs
#[cfg(feature = "unstable_discord_api")]
#[derive(Clone, Debug)]
pub struct AuthorizingIntegrationOwners(Vec<AuthorizingIntegrationOwner>);
```
and then implementing the (de)serialization for that.

Reply:
What situation causes multiple authorizing integration owners?

Reply:
Having the app installed as a user app aswell as on a guild you're on.

Reply:
Okay, done.